src/java.base/share/classes/java/nio/file/Files.java

```diff
@@ -2938,7 +2938,12 @@ public static BufferedReader newBufferedReader(Path path) throws IOException {
      * a size of {@code 0} if it exists.
      *
      * <p> The {@code Writer} methods to write text throw {@code IOException}
-     * if the text cannot be encoded using the specified charset.
+     * if the text cannot be encoded using the specified charset. Such coding
+     * exceptions will not occur until characters are flushed to the underlying
+     * stream either because the internal buffer was filled during writing, or
+     * {@linkplain BufferedWriter#close close()} or
+     * {@linkplain BufferedWriter#flush flush()}
+     * was invoked explicitly on the returned {@code BufferedWriter}.
```

Comment:
There isn't an "underlying stream" here but "file" could work. Alternatively, maybe this would work:

"Due to buffering, an IOException due to an encoding error (unmappable-character or malformed-input) may be thrown when writing, flushing or closing the buffered writer."

Reply:
> There isn't an "underlying stream" here but "file" could work. Alternatively, maybe this would work:

Well there is a stream created internally but I guess that's an implementation detail we need not mention.

> "Due to buffering, an IOException due to an encoding error (unmappable-character or malformed-input) may be thrown when writing, flushing or closing the buffered writer."

Updated in 20369045bcba2fac3a34c5d6f7ac3f54e582eb3f.